src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1373,22 +1373,38 @@ private NamedTypeSymbol ConstructNamedTypeUnlessTypeArgumentOmitted(SyntaxNode t
         {
             if (typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument))
             {
-                // Note: lookup won't have reported this, since the arity was correct.
-                // CONSIDER: the text of this error message makes sense, but we might want to add a separate code.
-                Error(diagnostics, ErrorCode.ERR_BadArity, typeSyntax, type, MessageID.IDS_SK_TYPE.Localize(), typeArgumentsSyntax.Count);
+                if (this.IsInsideNameof)
+                {
+                    // Inside a nameof an open-generic type is acceptable.  Fall through and bind the remainder accordingly.
+                    CheckFeatureAvailability(typeSyntax, MessageID.IDS_FeatureUnboundGenericTypesInNameof, diagnostics);
+
+                    // From the spec:
+                    //
+                    // Member lookup on an unbound type expression will be performed the same way as for a `this`
+                    // expression within that type declaration.
+                    //
+                    // So we want to just return the originating type symbol as is (e.g. List<T> in nameof(List<>)).
+                    // This is distinctly different than how typeof(List<>) works, where it returns an unbound generic
+                    // type.
```

Comment:
For the question of return a definition vs. "unbound" (`SubstitutedNamedTypeSymbol._unbound`) and the behavior of GetMembers, I think this needs a bit more investigation.
I like what you have in the spec and the we've aligned the implementation to return the definition, but something feels off. Why didn't we return the definition for unbound in other scenarios? (why did we need the `_unbound` flag?)
It's not clear to me whether the GetMembers behavior that you pointed out is a feature (we really don't want to return non-type members from unbound because of reason X), or an optimization (we don't need to bother because no scenario needs non-type members on unbound types).

Reply:
Who does this investigation?

Reply:
Discussed with @AlekseyTs and team at standup today. 
TL;DR: returning the definition (rather than an "unbound" symbol) is fine :-)

For purpose of `typeof` we need to differentiate between an unbound type and the definition, because it would impact what IL we emit. Also as we'd discussed, for `typeof` we ever only needed nested types (not other kinds of members).  
But for purpose of `nameof` we don't need that, the type definition is fine, since we'll only need its name and we do want both type and non-type members.  
This choice will be visible in the symbol from the semantic model, but the difference is minor.

Also, FWIW, the [standard](https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/types.md#844-bound-and-unbound-types) supports using the definition for unbound types in general:
> The term unbound type refers to a non-generic type or an unbound generic type. The term bound type refers to a non-generic type or a constructed type.
> An unbound type refers to the entity declared by a type declaration. An unbound generic type is not itself a type, and cannot be used as the type of a variable, argument or return value, or as a base type.